src/common/transport/transport-client.js

```diff
@@ -0,0 +1,80 @@
+import { TransportClientError } from '../errors';
+import rp from 'request-promise';
```

Comment:
What are your thoughts on either:

1. Using request-promise-native (dropping bluebird) and allowing for some of that async/await generator stuff.
2. Using axios and making this a domo-js-sdk. (Can run on the browser or node or other JS runtimes)